tests/tasks/monte_carlo/test_monte_carlo_lineage.py

```diff
@@ -0,0 +1,303 @@
+
+
+@responses.activate
+def test_monte_carlo_create_or_update_lineage():
+    api_key_id = "your_api_key_id"
+    api_token = "your_api_token"
+    source = dict(
+        node_name="test_node",
+        object_id="test_node",
+        object_type="table",
+        resource_name="test_system",
+    )
+    destination = dict(
+        node_name="test_dwh_table",
+        object_id="test_dwh_table",
+        object_type="table",
+        resource_name="test_dwh",
+    )
+    # source node:
+    responses.add(
+        responses.POST,
+        MONTE_CARLO_API_URL,
+        status=200,
+        json={
+            "data": {"createOrUpdateLineageNode": {"node": {"mcon": "MCON++123456789"}}}
+        },
+    )
+    # destination node:
+    responses.add(
+        responses.POST,
+        MONTE_CARLO_API_URL,
+        status=200,
+        json={
+            "data": {"createOrUpdateLineageNode": {"node": {"mcon": "MCON++987654321"}}}
+        },
+    )
+    # lineage edge:
+    responses.add(
+        responses.POST,
+        MONTE_CARLO_API_URL,
+        status=200,
+        json={
+            "data": {
+                "createOrUpdateLineageEdge": {
+                    "edge": {
+                        "edgeId": "f556899cfd39993517d7fc262f3d22d759fe94bd54324c5d89f4be48333dcab7",
+                    }
+                }
+            }
+        },
+    )
+    result = MonteCarloCreateOrUpdateLineage(
+        source, destination, api_key_id, api_token, prefect_context_tag=False
+    ).run()
+    assert result == "f556899cfd39993517d7fc262f3d22d759fe94bd54324c5d89f4be48333dcab7"
+
+
+@pytest.mark.parametrize(
+    "source",
+    [
+        dict(
+            object_id="test_node",
+            object_type="table",
+            resource_name="test_system",
+        ),
+        dict(
+            node_name="test_node",
+            object_type="table",
+            resource_name="test_system",
+        ),
+        dict(
+            node_name="test_node",
+            object_id="test_node",
+            object_type="table",
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "destination",
+    [
+        dict(
+            object_id="test_node",
+            object_type="table",
+            resource_name="test_system",
+        ),
+        dict(
+            node_name="test_node",
+            object_type="table",
+            resource_name="test_system",
+        ),
+        dict(
+            node_name="test_node",
+            object_id="test_node",
+            object_type="table",
+        ),
+    ],
+)
+def test_monte_carlo_create_or_update_lineage_raises_with_missing_attributes(
+    source, destination
+):
+    api_key_id = "your_api_key_id"
+    api_token = "your_api_token"
+    with pytest.raises(ValueError) as exc:
+        MonteCarloCreateOrUpdateLineage(
+            source, destination, api_key_id, api_token, prefect_context_tag=False
+        ).run()
+    assert "in both source and destination" in str(exc)
```

Comment:
You can use `match` to specify a regex substring to check for
```suggestion
    with pytest.raises(ValueError, match="in both source and destination"):
        MonteCarloCreateOrUpdateLineage(
            source, destination, api_key_id, api_token, prefect_context_tag=False
        ).run()
```

Reply:
Nice, TMYK! I updated this in all MC tests that use this pattern